src/GraphBLAS-sharp.Backend/Objects/Matrix.fs

```diff
@@ -181,3 +181,36 @@ type ClMatrix<'a when 'a: struct> =
         | ClMatrix.COO matrix -> matrix.NNZ
         | ClMatrix.CSC matrix -> matrix.NNZ
         | ClMatrix.LIL matrix -> matrix.NNZ
+
+/// <summary>
+/// Represents an abstraction over matrix, which is converted to correct format for PageRank algorithm
+/// </summary>
+type PageRankMatrix<'a when 'a: struct> =
```

Comment:
Great. But it needs to be even stronger. Abstract types are needed here. Like here in the first example.

https://okmij.org/ftp/Computation/lightweight-static-guarantees.html

I was told that this can be done using signatures. The idea is that we should not allow anyone to use this type unless they have passed some kind of verification. It seems to me that this also applies to disjoint matrices, @artemiipatov .

Of course, this can be disabled during benchmarking.

In the same way, we will add another functional and at the same time zero abstraction. This is exactly what is needed on FHPNC

Reply:
Signatures are not working?